test/e2e/network/dual_stack_host_ips.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package network
+
+import (
+	"context"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubefeatures "k8s.io/kubernetes/pkg/features"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	"k8s.io/kubernetes/test/e2e/network/common"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+	netutils "k8s.io/utils/net"
+)
+
+var _ = common.SIGDescribe("Dual Stack [Feature:PodHostIPs]", func() {
+	f := framework.NewDefaultFramework("dualstack")
+
+	ginkgo.It("should create pod, add host ips is empty", func() {
+		e2eskipper.SkipIfFeatureGateEnabled(kubefeatures.PodHostIPs)
+
+		podName := "pod-dualstack-host-ips"
+
+		pod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   podName,
+				Labels: map[string]string{"test": "dualstack-host-ips"},
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:  "dualstack-host-ips",
+						Image: imageutils.GetE2EImage(imageutils.Agnhost),
+					},
+				},
+			},
+		}
+
+		ginkgo.By("submitting the pod to kubernetes")
+		podClient := f.PodClient()
+		p := podClient.CreateSync(pod)
+
+		gomega.Expect(p.Status.HostIP).ShouldNot(gomega.BeEquivalentTo(""))
```

Comment:
we should do more on assertion, you can get the host-ip from the API and verify that is the same, just checking that is not empty is not enough

Reply:
`HostIP` should not be the focus of this feature attention, should more consider whether `HostIPs` meet?

Reply:
ah, yeah, I confused it with `HostIPs` , sorry